core/templates/rid_owner.h

```diff
@@ -350,24 +350,24 @@ class RID_Alloc : public RID_AllocBase {
 				if (validator & 0x80000000) {
 					continue; //uninitialized
 				}
-				if (validator != 0xFFFFFFFF) {
+				if (chunks != nullptr && validator != 0xFFFFFFFF) {
 					chunks[i / elements_in_chunk][i % elements_in_chunk].~T();
 				}
 			}
 		}
 
 		uint32_t chunk_count = max_alloc / elements_in_chunk;
+		if (chunks != nullptr) {
```

Comment:
This looks like you're deleting the array of pointers before deleting the actual pointers? (Sorry that it's not all the lines of change haven't used this feature before)

It looks to me like you'll get a segfault since when the pointers are iterated over and freed the containing arrays will be freed already 

Instead you need to move the check to encompass the iteration and the final free

Reply:
Thank you for your comment. I noticed the discrepancy there and moved the null check for freeing memory after the pointers are iterated over.

Reply:
This won't prevent problems as the pointer is accessed in the loop, the freeing needs to happen after, but the check prior no?

Reply:
Also no need to check against nullptr, the check for non-zero is standard behaviour as far as I know and used across the project, that is not the problem

Reply:
Oh yes I realized the count is zero if allocated so doesn't cause a problem, my bad